src/compute/src/sink/persist_sink.rs

```diff
@@ -745,6 +745,9 @@ where
                     // spend a lot of time "consolidating" the same updates
                     // over and over again, with no changes.
                     consolidate_updates(&mut correction);
+                    if correction.len() < correction.capacity() / 4 {
```

Comment:
copy the PR description in here as a comment, please!

Reply:
Fixed up in the subsequent commit.